pensieve/config.py

```diff
@@ -163,18 +252,49 @@ class MetricDefinition:
     name: str  # implicit in configuration
     select_expression: str
     data_source: DataSourceReference
+    statistics: Dict[str, Dict[str, Any]]
+    pre_treatments: List[PreTreatmentReference] = attr.Factory(list)
 
-    def resolve(self, spec: "AnalysisSpec") -> mozanalysis.metrics.Metric:
+    def resolve(
+        self, spec: "AnalysisSpec", experimenter: pensieve.experimenter.Experiment
+    ) -> List[Summary]:
         select_expression = _metrics_environment.from_string(self.select_expression).render()
 
-        return mozanalysis.metrics.Metric(
+        metric = mozanalysis.metrics.Metric(
             name=self.name,
             data_source=self.data_source.resolve(spec),
             select_expr=select_expression,
         )
 
+        metrics_with_treatments = []
+
+        for statistic_name, params in self.statistics.items():
+            for statistic in Statistic.__subclasses__():
+                if statistic.name() == statistic_name:
+                    pre_treatments = [pt.resolve for pt in self.pre_treatments]
+
+                    if "ref_branch_label" not in params:
```

Comment:
So unless `ref_branch_label` is explicitly set in the config we'll use the control branch from experimenter. Or else it's just the default `"control"` branch